__tests__/app.test.js

```diff
@@ -202,4 +202,148 @@ describe("api/articles/:article_id/comments", () => {
     })
   })
 
-}); // end of "api/articles/:article_id/comments" testing
\ No newline at end of file
+}); // end of "GET api/articles/:article_id/comments" testing
+
+describe("POST /api/articles/:article_id/comments", () => {
+
+  test("responds with 201 status for successful requests", () => {
+
+   return request(app)
+    .post("/api/articles/5/comments")
+    .send( {username: "butter_bridge", body: "this is a test"} )
+    .expect(201)
+  })
+
+  test("responds with newly added comment", () => {
+
+   return request(app)
+    .post("/api/articles/5/comments")
+    .send( {username: "icellusedkars", body: "also a test"} )
+    .then( data => {
+      expect(data.body.comment_added).toEqual(
+        expect.objectContaining({
+          comment_id: 19,
+          body: "also a test",
+          article_id: 5,
+          author: "icellusedkars",
+          votes: 0,
+          created_at: expect.any(String),
+        })
+      )
+    })
+  })
+
+  test("error passed to handlers, responds with 404 when username violates foreign key constraint", () => {
+
+    return request(app)
+     .post("/api/articles/3/comments")
+     .send( { username: "isaac", body: "this shouldn't work" } )
+     .expect(404)
+     .then( data => {
+      expect(data.body.msg).toBe("not found");
+     })
+
+  })
+
+  test("responds with 404 and error if key is not present in articles table", () => {
+
+    return request(app)
+      .post("/api/articles/999/comments")
+      .send( {username: "icellusedkars", body: "also shouldn't work"} )
+      .expect(404)
+      .then( data => {
+      expect(data.body.msg).toBe("not found");
+      });
+  })
+
+  test("responds with 400 and error if username has not been defined", () => {
+
+    return request(app)
+    .post("/api/articles/5/comments")
+    .send( {body: "this is a test"} )
+    .expect(400)
+    .then( data => {
+      expect(data.body.msg).toBe("incomplete request, provide username")
+   })
+  })
+
+  test("responds with 400 and error if body has not been defined", () => {
+
+    return request(app)
+    .post("/api/articles/5/comments")
+    .send( {username: "icellusedkars"} )
+    .expect(400)
+    .then( data => {
+      expect(data.body.msg).toBe("incomplete request, please add a comment")
+   })
+  })
```

Comment:
Testing for the POST is brilliant, one optional test you can add is that it still completes the post even if there are additional unnecessary keys in the request body